bool rotate_cw
bool rotate_ccw